setup.py

```diff
@@ -26,6 +30,17 @@
     package_data.setdefault('setuptools.command', []).extend(['*.xml'])
 
 
+def _get_version() -> str:
+    cmd = ["git", "describe", "--abbrev", "--match", "v?[0-9]*", "--dirty"]
+    try:
+        version = subprocess.check_output(cmd, encoding="utf-8")
+        return _normalization.best_effort_version(version, "{safe}.dev+{sanitized}")
+    except subprocess.CalledProcessError:  # e.g.: git not installed or history missing
+        with open("NEWS.rst", encoding="utf-8") as fp:
+            match = re.search(r"v\d+\.\d+.\d+", fp.read())  # latest version
+            return f"{match[0] if match else '0.0.0'}.dev+{time.strftime('%Y%m%d')}"
```

Comment:
`%Y%m%d` can be changed to more specific time to improve monotonicity... Please let me know if a more specific tag would be better.

Reply:
BTW, the `except` is the code path that takes effect when git tag version is not available (tarballs or shallow clones).

I have considered checking if a `PKG-INFO`/`setuptools*.dist-info/METADATA` file is available but that would increase the complexity of this function... Please let me know if this more complex approach should be followed...

Reply:
OK, it may be necessary to check for `PKG-INFO`, because otherwise we may see the following behaviour when running `python -m build`:

```console
$ python -m build
Successfully built setuptools-78.1.2.tar.gz and setuptools-78.1.2.dev0+20250416-py3-none-any.whl
```

Reply:
> OK, it may be necessary to check for `PKG-INFO`, because otherwise we may see the following behaviour when running `python -m build`:

So I inevitably had to add some lines to deal with this problem:

https://github.com/pypa/setuptools/pull/4948/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R39-R42

It does increase the complexity of the solution